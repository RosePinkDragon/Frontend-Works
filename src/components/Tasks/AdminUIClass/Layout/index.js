import React from "react";
import userLogo from "images/userLogo.png";
import "styles/Layout.css";
import { sidebarData } from "utils/layoutData";
import SidebarSubMenu from "./SidebarSubMenu";
import { connect } from "react-redux";
import { logOut } from "../redux/actions/loginActions";

class Layout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isActive: false,
    };
  }

  handleLogOut = () => {
    this.props.logOut();
  };

  render() {
    return (
      <>
        <header>
          <nav>
            <h1>App Name</h1>
            <div>
              <img className="logo" src={userLogo} alt="" />
              <p>Moheb Dabilkar</p>
            </div>
            <div
              className={`burger-btn ${this.state.isActive ? "active" : ""}`}
              onClick={() => this.setState({ isActive: !this.state.isActive })}
            />
          </nav>
        </header>
        <div className={`sidebar ${this.state.isActive ? "active" : ""}`}>
          <div className="sidebarUserInfo">
            <img className="logo" src={userLogo} alt="" />
            <p>Moheb Dabilkar</p>
          </div>
          <div className="components">
            {sidebarData.map((data, idx) => (
              <SidebarSubMenu key={idx} data={data} />
            ))}
          </div>
          <div>
            <button className="btn logOut" onClick={() => this.handleLogOut()}>
              LogOut
            </button>
          </div>
        </div>
        <div className="bodyWrap">{this.props.children}</div>
      </>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return { logged: state.logged };
};

const mapDispatchToProps = (dispatch) => {
  return {
    logOut: () => {
      dispatch(logOut());
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
