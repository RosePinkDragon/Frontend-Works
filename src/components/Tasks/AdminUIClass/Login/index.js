import React from "react";
import { connect } from "react-redux";
import { logIn } from "../redux/actions/loginActions";
import "styles/login.css";

class LoginScreen extends React.Component {
  handleLogIn = (e) => {
    e.preventDefault();
    this.props.logIn();
  };

  render() {
    return (
      <div className="login">
        <form onSubmit={(e) => this.handleLogIn(e)}>
          <div className="formElement">
            <label htmlFor="email">Email</label>
            <input type="email" name="email" id="email" required />
          </div>
          <div className="formElement">
            <label htmlFor="password">Password</label>
            <input type="password" name="password" id="password" required />
          </div>
          <button className="btn">login</button>
        </form>
      </div>
    );
  }
}

// const mapStateToProps = (state, ownProps) => {
//   return;
// };

const mapDispatchToProps = (dispatch) => {
  return {
    logIn: () => {
      dispatch(logIn());
    },
  };
};

export default connect("", mapDispatchToProps)(LoginScreen);
