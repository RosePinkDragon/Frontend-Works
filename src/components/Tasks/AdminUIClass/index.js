import React from "react";
import propTypes from "prop-types";
import Layout from "components/Tasks/AdminUIClass/Layout";
import LoginScreen from "components/Tasks/AdminUIClass/Login/";
import Chart from "./Home/components/Chart";
import ProductCards from "./Home/components/ProductCards";
import { connect } from "react-redux";
import "styles/App.css";

// ?? data is used to store external data to reuse.
const mapStateToProps = (props) => {
  return {
    status: props.loginReducers.logged,
  };
};

class AdminUIClass extends React.Component {
  render() {
    return (
      <>
        {this.props.status ? (
          <Layout>
            <ProductCards />
            <Chart />
          </Layout>
        ) : (
          <LoginScreen />
        )}
      </>
    );
  }
}

// ?? we can also check for prop types as well
AdminUIClass.propTypes = {
  // logged: propTypes.bool,
  defProps: propTypes.object,
};

// ?? we can pass default props if the props as not passed
AdminUIClass.defaultProps = {
  defProps: {
    This: "is a def prop",
  },
};

export default connect(mapStateToProps)(AdminUIClass);
