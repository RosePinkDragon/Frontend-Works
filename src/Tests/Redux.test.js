import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import AdminUIClass from "components/Tasks/AdminUIClass";
import { logIn } from "components/Tasks/AdminUIClass/redux/actions/loginActions";

const mockStore = configureStore([thunk]);

describe("My Renders the expected page", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      loginReducers: {
        logged: false,
      },
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <AdminUIClass />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render AdminUI", () => {
    store = mockStore({
      loginReducers: {
        logged: true,
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <AdminUIClass />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should dispatch an action on button click", () => {
    renderer.act(() => {
      component.root
        .findByType("form")
        .props.onSubmit({ preventDefault: jest.fn() });
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(logIn());
  });
});
