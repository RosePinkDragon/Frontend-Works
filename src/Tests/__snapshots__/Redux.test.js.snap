// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Renders the expected page should render AdminUI 1`] = `
Array [
  <header>
    <nav>
      <h1>
        App Name
      </h1>
      <div>
        <img
          alt=""
          className="logo"
          src="userLogo.png"
        />
        <p>
          Moheb Dabilkar
        </p>
      </div>
      <div
        className="burger-btn "
        onClick={[Function]}
      />
    </nav>
  </header>,
  <div
    className="sidebar "
  >
    <div
      className="sidebarUserInfo"
    >
      <img
        alt=""
        className="logo"
        src="userLogo.png"
      />
      <p>
        Moheb Dabilkar
      </p>
    </div>
    <div
      className="components"
    >
      <div
        className="subMenu"
      >
        <p
          onClick={[Function]}
        >
          Admin Menu
        </p>
      </div>
      
      <div
        className="subMenu"
      >
        <p
          onClick={[Function]}
        >
          User Menu
        </p>
      </div>
      
      <div
        className="subMenu"
      >
        <p
          onClick={[Function]}
        >
          Client Menu
        </p>
      </div>
      
      <div
        className="subMenu"
      >
        <p
          onClick={[Function]}
        >
          Dev Menu
        </p>
      </div>
      
    </div>
    <div>
      <button
        className="btn logOut"
        onClick={[Function]}
      >
        LogOut
      </button>
    </div>
  </div>,
  <div
    className="bodyWrap"
  >
    <div
      className="cardWrap"
    >
      <div
        className="card"
      >
        <h3>
          Passed Products
        </h3>
        <h5>
          6
        </h5>
      </div>
      <div
        className="card"
      >
        <h3>
          Upcoming Products
        </h3>
        <h5>
          6
        </h5>
      </div>
      <div
        className="card"
      >
        <h3>
          Total Products
        </h3>
        <h5>
          6
        </h5>
      </div>
    </div>
    <div
      className="chartWrap"
    >
      <div>
        Loading...
      </div>
      <canvas
        className="chart doughnut"
        data-testid="canvas"
        height={150}
        onClick={[Function]}
        role="img"
        width={300}
      />
      <canvas
        className="chart bar"
        data-testid="canvas"
        height={150}
        onClick={[Function]}
        role="img"
        width={300}
      />
    </div>
  </div>,
]
`;

exports[`My Renders the expected page should render with given state from Redux store 1`] = `
<div
  className="login"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="formElement"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="formElement"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="btn"
    >
      login
    </button>
  </form>
</div>
`;
